### PR TITLE
fix opengl test error, add the condition

### PR DESCRIPTION
**Pull request by jianglong0156**
fix opengl test error, add the condition, when the data type is samewith long, it will reset data
relate pr:#13449
fix open gl show error

### CODE REVIEW

File: cocos/scripting/js-bindings/manual/js_manual_conversions.cpp [hunk 1]
```diff
@@ -187,8 +187,13 @@ bool jsval_to_int( JSContext *cx, JS::HandleValue vp, int *ret )
     // Since this is called to cast uint64 to uint32,
     // it is needed to initialize the value to 0 first
 #ifdef __LP64__
-    long *tmp = (long*)ret;
-    *tmp = 0;
+    // only unsigned int size is 8Bit, same with the long, it's need to reset the data.
```
Comment (pandamicro) [inline cocos/scripting/js-bindings/manual/js_manual_conversions.cpp]: When `int` size is 8 Bit (same as `long`), the following operation is needed

File: cocos/scripting/js-bindings/manual/js_manual_conversions.cpp [hunk 2]
```diff
@@ -243,8 +248,13 @@ bool jsval_to_uint( JSContext *cx, JS::HandleValue vp, unsigned int *ret )
...
```
Comment (pandamicro) [inline cocos/scripting/js-bindings/manual/js_manual_conversions.cpp]: When `unsigned int` size is 8 Bit (same as `long`), the following operation is needed

**Comment by pandamicro** [comment]
Please squash all commits into one

**Comment by jianglong0156** [comment]
other squash pr:https://github.com/cocos2d/cocos2d-x/pull/13487
close this pr

**Pull request closed without merging**